mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -276,6 +277,177 @@ def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.D
         return ret
 
     def acquisition_function(self, configurations: pd.DataFrame,
-                             context: Optional[pd.DataFrame] = None) -> Callable:
+                             context: Optional[pd.DataFrame] = None) -> npt.NDArray:
         # This seems actually non-trivial to get out of skopt, so maybe we actually shouldn't implement this.
         raise NotImplementedError()
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
+    """ Wrapper class for SMAC based Bayesian optimization.
+
+    Parameters
+    ----------
+    parameter_space : ConfigSpace.ConfigurationSpace
+        The parameter space to optimize.
+    """
+    def __init__(
+        self,
+        parameter_space: ConfigSpace.ConfigurationSpace,
+        space_adapter: Optional[BaseSpaceAdapter] = None,
+        seed: Optional[int] = None,
+        run_name: str = 'smac',
+        output_directory: str = 'smac3_output',
+        n_random_init: Optional[int] = 10,
+        n_random_probability: Optional[float] = 0.1,
+        n_workers: int = 1,
+    ):
+        super().__init__(parameter_space, space_adapter)
+
+        from smac import HyperparameterOptimizationFacade as Optimizer_Smac # pylint: disable=import-outside-toplevel
+        from smac import Scenario # pylint: disable=import-outside-toplevel
+        from smac.intensifier.abstract_intensifier import AbstractIntensifier # pylint: disable=import-outside-toplevel
+        from smac.initial_design import LatinHypercubeInitialDesign # pylint: disable=import-outside-toplevel
+        from smac.main.config_selector import ConfigSelector # pylint: disable=import-outside-toplevel
+        from smac.random_design.probability_design import ProbabilityRandomDesign # pylint: disable=import-outside-toplevel
+        from smac.runhistory import TrialInfo # pylint: disable=import-outside-toplevel
+
+        self.seed: int = seed or 0
```

Comment:
why not have `seed: int = 0` in `__init__` parameters?

Reply:
actually, no - keep `seed: Optional[int] = None` - but maybe do NOT make the seed 0 by default - this effectively makes our optimizer deterministic even if we do not specify the seed (if I understand it correctly).

Reply:
Actually, SMAC does not accept a `None` value for its `seed` argument -- it is set to `0` by default and can only take int values.

I think there is value to having a `None` value: e.g., we can set the `seed` to -1, which results in SMAC generating a random seed internally.

Reply:
Any comments on the above? @motus @bpkroth

Reply:
I agree with @motus, I think a random seed is probably a better default.

I'm not sure I see a point in accepting `None` to the argument though.

I think

```python
    seed: int = -1,
```

is probably fine as the argument to `__init__` and let SMAC throw an error if `None` is passed for some reason.

Reply:
So the default for SMAC is `0` to make experiments reproducible. `None` will throw an error.

FYI https://automl.github.io/SMAC3/main/api/smac.scenario.html#smac.scenario.Scenario

Reply:
Eh, I'm fine with either I guess.

Let's do this:

```python
   ...
   seed: Optional[int] = 0,
   ...

   self.seed = -1 if seed is None else seed
```